app/views/handlers/list_remove_question.py

```diff
@@ -3,46 +3,51 @@
 
 
 class ListRemoveQuestion(ListAction):
-    def is_location_valid(self):
+    def is_location_valid(self) -> bool:
         list_item_doesnt_exist = (
             self._current_location.list_item_id
             not in self._questionnaire_store.list_store[
-                self._current_location.list_name
+                # type ignore: list_name will exist within the remove block
+                self._current_location.list_name  # type: ignore
             ].items
         )
         is_primary = (
             self._questionnaire_store.list_store[
-                self._current_location.list_name
+                # type ignore: list_name will exist within the remove block
+                self._current_location.list_name  # type: ignore
             ].primary_person
             == self._current_location.list_item_id
         )
         if not super().is_location_valid() or list_item_doesnt_exist or is_primary:
             return False
         return True
 
-    def handle_post(self):
+    def handle_post(self) -> None:
         answer_action = self._get_answer_action()
 
         if answer_action and answer_action["type"] == "RemoveListItemAndAnswers":
             list_name = self.parent_block["for_list"]
             self.questionnaire_store_updater.remove_list_item_and_answers(
-                list_name, self._current_location.list_item_id
+                # type ignore: list_item_id will exist within the remove block
+                list_name,
+                self._current_location.list_item_id,  # type: ignore
             )
             self.capture_dependent_sections_for_list(self.parent_block["for_list"])
 
         return super().handle_post()
 
     def individual_response_enabled(self) -> bool:
-        return (
-            self.parent_block["for_list"] == self._schema.get_individual_response_list()
-        )
+        if self.parent_block["for_list"] == self._schema.get_individual_response_list():
```

Comment:
Also why have we reverted from the simplified version?

Reply:
It was throwing a "Returning Any from function declared to return bool" error

Reply:
since mypy doesn't know what the types for the immutable dictionary are, and so doesn't know how the `__eq__ `method is implemented, but casting for the sake of type hinting is avoided, but type hinting documentation states to only type ignore where its unavoidable, could this be a middle ground?
```
list_name: str = self.parent_block["for_list"]
return list_name == self._schema.get_individual_response_list()
```
I'm wondering if we need to document some sort of agreement on this type of scenario where it's obviously a boolean, and the above is longer than the simplified version, and we _can_ avoid a type ignore, but maybe don't want to in this case?

Reply:
Yeah probably need to discuss this, I added this comment thinking more about not amending code for the sake of type hints, but can see where you are coming from with this hint technically being avoidable . Maybe one to raise in a tech session then and as a team we can agree whether we need to update the doc.

Reply:
Will make a note about updating this in tech session and resolve for now